### PR TITLE
Fix broken DLL imports on Windows+Conda+Python>=3.8

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
2 different issues with the same symptom (`DLL could not be found`)

### Issue 1
Python >= 3.8 changed the DLL import mechanism on windows, but conda have disabled this change and still require explicitly changing PATH rather than calling os.add_dll_directory. This was an issue on my local windows dev machine.

### Issue 2
DLLs were not getting copied to a directory where they could be found by the conda build script. I have moved the install location of the DLLs into the `scipp` library so that they can be found along with the library. This was the issue that was causing the windows CI builds to fail; with this change, the CI is now passing.

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
Try reenabling the commented sections in `.azure-pipelines/*yml` (in particular `pull_request.yml`) to see if this makes the Windows build pass.

File: python/src/scipp/__init__.py [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
+    # don't respect os.add_dll_path(...), which is python's mechanism for setting DLL search directories. Instead we
+    # need to explicitly add it to the PATH environment variable. For non-conda versions of python we want to keep using
+    # the usual python mechanism.
+    #
+    # This is probably due to the following patch in conda-built versions of python:
+    # https://github.com/conda-forge/python-feedstock/blob/289b2a8017ddd000896e525f18867f4caacec6f2/recipe/patches/0020-Add-CondaEcosystemModifyDllSearchPath.patch
+    #
+    import pathlib
+    dll_directory = os.path.abspath(
+        os.path.join(pathlib.Path(__file__).parent.resolve(), "..", "bin"))
```
Comment (jl-wynen) [inline python/src/scipp/__init__.py:20]: Using `__file__` could cause problems in a non-conda build if the package gets zipped. Not sure if that is possible with scipp, but in any case, better use https://docs.python.org/3/library/importlib.html#importlib.resources.path

Also, it is a bit strange to combine `os.path` and `Path` like this, you could just use `(path_from_resources.parent/'bin').resolve()`

**Review by jl-wynen — COMMENTED** [review]
Nice find!

**Comment by SimonHeybrock** [comment]
Doesn't work? :(

**Comment by Tom-Willemsen** [comment]
> Doesn't work? :(

Fixed now in latest commit; there were two distinct issues here which happened to have the same symptoms.

**Review by SimonHeybrock — APPROVED** [review]

**Comment by SimonHeybrock** [comment]
@Tom-Willemsen I now run into an issue, scippneutron does not find the DLLs, after they have been moved: https://dev.azure.com/scipp/scippneutron/_build/results?buildId=7299&view=logs&j=d5c58a2e-1ae4-501e-e421-b21c980eb30a&t=fcbe8920-cf97-5746-799c-eac6f4530465. Note that this is cmake not finding them, not Python. Do we need to change the paths in another way, or does this need fixing on the cmake config side?

**Pull request merged**